### ISSUE TITLE
✨ Making experimental factory functions of `NotEmptyList.Companion` available on Java

### ISSUE THREAD

**Issue by LVMVRQUXL** [kotools/types#497]
## 📝 Description

We would like to make the experimental factory functions of the `NotEmptyList.Companion` type with the following signature available for Java users:

```kotlin
fun <E> NotEmptyList.Companion.create(collection: Collection<E>): NotEmptyList<E>
fun <E> NotEmptyList.Companion.createOrNull(collection: Collection<E>): NotEmptyList<E>?

fun <E> NotEmptyList.Companion.of(head: E, vararg tail: E): NotEmptyList<E>
```

This means that their compatibility should be tested and their documentation should indicate their usage from Java code.

## 🔗 Dependencies

This issue is blocked by the following ones:
- [x] #352 
- [x] #435 

## ✅ Checklist

- [ ] Make the `create(Collection<E>)` function available for Java users, [test] its compatibility, update the public [API binaries] and update the [unreleased changelog].
- [ ] Make the `createOrNull(Collection<E>)` function available for Java users, [test] its compatibility, update the public [API binaries] and update the [unreleased changelog].
- [ ] Make the `of(E, vararg E)` function available for Java users, [test] its compatibility, update the public [API binaries] and update the [unreleased changelog].
- [ ] Close this issue as completed and update tracking ones if present.

[api binaries]: https://github.com/kotools/types/blob/main/CONTRIBUTING.md#checking-the-api-binaries
[test]: https://github.com/kotools/types/blob/main/CONTRIBUTING.md#running-tests
[unreleased changelog]: https://github.com/kotools/types/blob/main/CHANGELOG.md#unreleased

**Comment by LVMVRQUXL** [comment]
Another issue for stabilizing these functions will be created while resolving #619.
